cypress/GenerateCypressTestSpecs.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.  
+ * Licensed under the MIT License.
+*/
+
+// ----------------------------------------------------------------------
+// This code is intended to be run via NODE.js.
+// It is intended to be started automatically whenever test cases 
+// are being initialized to run by Cypress.
+//
+// This code parses all .js files in the cypress/tests folder.
+// It is looking for lines in the files that look like this:
+//  Cypress.TestCase(testGroupName, testDescription, testFunction)
+//
+// It uses that data to create:
+//   1) The group subfolders within "cypress/integration".
+//   2) An individual test spec file for each, in that same folder as 
+//      expected by Cypress.
+//   3) The test list: masterListOfAllTestCases
+// ----------------------------------------------------------------------
+
+var fs = require('fs')
+
+const pathToTestFiles = './cypress/tests'
+var fullTestList = []
+
+main();
+WatchAndRun('./cypress/tests', main)
+
+// ----------------------------------------------------------------------
+
+function WatchAndRun(pathToWatch, functionToRun) {
+  var timeoutHandle;
+  var timeOfLastChange;
+  
+  fs.watch(pathToWatch, (eventType, filename) => {
+    var now = new Date();
+    if (timeOfLastChange && now - timeOfLastChange < 1000) { clearTimeout(timeoutHandle); }
+    timeOfLastChange = now;
+    timeoutHandle = setTimeout(functionToRun, 1000)
+  });
+}
+
+// ----------------------------------------------------------------------
+
+function main() {
+  try {
+    // Get the test list file that we MIGHT need to modify.
+    const pathToTestList = './cypress/TestLists.js';
+    var testList_jsContents = fs.readFileSync(pathToTestList, 'ascii');
+    // console.log(testList_jsContents)
+    // console.log()
+
+    // Get the list of test js files.
+    var testFileDirectory = fs.readdirSync(pathToTestFiles);
+    // console.log(testFileDirectory);
+    // console.log()
+
+    // Parse each test js file.
+    testFileDirectory.forEach(fileName => { ParseTestJsFile(fileName); });
+
+    // Finalize our findings by re-writing out the TestList.js file ONLY if it has changed.
+    var index = testList_jsContents.indexOf('// ************ Generated Code Beyond this Point *************************');
+    var newFileContents =
+      '// ************ Generated Code Beyond this Point *************************\r\n' +
```

Comment:
Not required to change but you can use backtick (`) to create multiline strings. I find it more readable because there isn't extra characters in either side and definitely more editable.

```
const newFileContents = `
// ************ Generated Code Beyond this Point *************************
// Do NOT manually alter this file from this point onwards.
// Any changes you make will be overridden at runtime.
`

Only odd thing is it has to be left justified or the whitespace will be included.

Reply:
EXCELLENT Suggestion! Keep em coming. (I missed that in the last CR)

Reply:
I wasn't aware you could do that.